plugins/com.google.cloud.tools.eclipse.test.util/src/com/google/cloud/tools/eclipse/test/util/project/ProjectUtils.java

```diff
@@ -105,24 +109,68 @@
 
     // wait for any post-import operations too
     waitUntilIdle();
+    assertFalse("imported projects have errors", projectsHaveErrors(projects));
 
     return projects;
   }
 
-  /** Wait for any spawned jobs to complete (e.g., validation jobs). */
-  public static void waitUntilIdle() {
-    while (!Job.getJobManager().isIdle()) {
-      Display display = Display.getCurrent();
-      if (display != null) {
-        while (display.readAndDispatch()) {
-          /* spin */
+  /**
+   * Return true if projects have errors; lists any found to <code>System.err</code>.
+   */
+  public static boolean projectsHaveErrors(Collection<IProject> projects) {
+    List<String> foundProblems = new ArrayList<>();
+    for (IProject project : projects) {
+      try {
+        IMarker[] problems = project.findMarkers(IMarker.PROBLEM, true /* includeSubtypes */,
+            IResource.DEPTH_INFINITE);
+        for (IMarker problem : problems) {
+          int severity = problem.getAttribute(IMarker.SEVERITY, IMarker.SEVERITY_INFO);
+          if (severity >= IMarker.SEVERITY_ERROR) {
+            foundProblems.add(formatProblem(problem));
+          }
         }
+      } catch (CoreException ex) {
+        // re-throw as we shouldn't have problems in tests
+        throw new RuntimeException(ex);
       }
-      try {
+    }
+    if (foundProblems.isEmpty()) {
+      return false;
+    }
+    System.err.println("Project errors:");
+    for (String problem : foundProblems) {
+      System.err.println(" " + problem);
```

Comment:
This is intended as diagnostics — I'd have saved some time trying to fix this if I could have dumped the workspace errors on the Travis build log.

Better suggestion?  I guess this could be `failIfBuildErrors()`, and then have it `fail()` with the stringified errors?